thrust/benchmarks/bench/equal/basic.cu

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <thrust/device_vector.h>
+#include <thrust/equal.h>
+#include <thrust/execution_policy.h>
+
+#include "nvbench_helper.cuh"
+
+template <typename T>
+static void benchmark(nvbench::state& state, nvbench::type_list<T>)
+{
+  const auto& name    = state.get_benchmark().get_name();
+  const auto elements = static_cast<std::size_t>(state.get_int64("Elements"));
+  thrust::device_vector<T> in(elements, T{1});
+  thrust::device_vector<T> b(elements, T{1});
+  caching_allocator_t alloc;
+
+  if (name == "first")
+  {
+    b.front() = T{2};
+  }
+  else if (name == "last")
+  {
+    b.back() = T{2};
+  }
+  else
+  {
+    const auto common_prefix   = state.get_float64("CommonPrefixRatio");
+    const auto different_elems = std::min(static_cast<std::size_t>(elements * common_prefix), elements);
+    thrust::fill(policy(alloc), b.begin() + different_elems, b.end(), T{2});
+  }
+
+  state.add_element_count(elements);
+  state.add_global_memory_reads<T>(2 * elements);
```

Comment:
I wonder whether reporting bandwidth actually makes sense, since `equal` has ample changes to return early.